### PR TITLE
java 17 upgrade

### PR DESCRIPTION
**Pull request by mlorek**
#2353

### CODE REVIEW

**Comment by teverett** [comment]
@mlorek thanks

File: java/java/JavaLexer.g4 [hunk 1]
```diff
@@ -97,6 +99,17 @@ TRANSITIVE:         'transitive';
...
 
+// Switch Expressions
+YIELD:              'yield';
+
+// Records
+RECORD:             'record';
+
+// Sealed Classes
+SEALED:             'sealed';
+PERMITS:            'permits';
```
Comment (KvanTTT) [inline java/java/JavaLexer.g4:110]: They all also can be identifiers. Please all `yield` `record`, `sealed`, `permits` to `identifier` rule.

Comment (mlorek) [reply java/java/JavaLexer.g4:110]: > They all also can be identifiers. Please all `yield` `record`, `sealed`, `permits` to `identifier` rule.

it's already there

Comment (KvanTTT) [reply java/java/JavaLexer.g4:110]: Sorry, I missed that. But `yield` is still missing and it's an acceptable identifier as well.

**Pull request merged**